tests/test_rabbitmq.py

```diff
@@ -464,6 +465,38 @@ def do_work():
             worker.stop()
 
 
+def test_rabbitmq_broker_retries_declaring_queues_when_declared_queues_is_gone(rabbitmq_broker):
+    executed = False
+
+    # I declare an actor
+    @dramatiq.actor(queue_name="flaky_queue")
+    def do_work():
+        nonlocal executed
+        executed = True
+
+    # Let worker to ensure_queue and consume message
+    worker = Worker(rabbitmq_broker, worker_threads=1)
+    worker.start()
+
+    # Check the queue is declared
+    rabbitmq_broker.channel.queue_declare(do_work.queue_name, passive=True)
+    # Let the queue go unexpectedly
+    rabbitmq_broker.channel.queue_delete(do_work.queue_name)
+    with pytest.raises(pika.exceptions.ChannelClosedByBroker):
+        rabbitmq_broker.channel.queue_declare(do_work.queue_name, passive=True)
+
+    # And I send that actor a message
+    do_work.send()
+    try:
+        rabbitmq_broker.join(do_work.queue_name, timeout=20000)
+        worker.join()
+
+        # Then the queue should eventually be declared and the message executed
+        assert executed
+    finally:
+        worker.stop()
```

Comment:
Looks like try/finally is redundant

```suggestion
    # And I send that actor a message
    do_work.send()
    rabbitmq_broker.join(do_work.queue_name, timeout=20000)
    worker.join()
    worker.stop()

    # Then the queue should eventually be declared and the message executed
    assert executed
```

Reply:
Be ensure to cover the exception then close the worker safely during the test. I reserve the try/finally scope. But I move the assert statement to the end of the test case.

Reply:
Be ensure to cover the exception then close the worker safely during the test. I reserve the try/finally scope. But I move the assert statement to the end of the test case.